sw/host/tests/rom/e2e_bootstrap_rma/src/main.rs

```diff
@@ -18,362 +21,322 @@ use opentitanlib::dif::lc_ctrl::{
+    )
+    .context("failed to reset with RMA_BOOTSTRAP strapping")?;
+
+    log::info!("Waiting for reset reasons over the console...");
+
+    // Strings expected from the console representing success or failure.
+    let expected_bitfield = u32::from(DifRstmgrResetInfo::Por) | u32::from(DifRstmgrResetInfo::Sw);
+    let exit_success =
+        Regex::new(format!("reset_info_bitfield: 0x{expected_bitfield:x}\r\n").as_str())
+            .context("failed to build regex")?;
+    let exit_failure =
+        Regex::new("reset_info_bitfield: 0x[0-9a-f]+\r\n").context("failed to build regex")?;
+
+    let mut console = UartConsole {
+        timeout: Some(CONSOLE_TIMEOUT),
+        exit_success: Some(exit_success),
+        exit_failure: Some(exit_failure),
+        ..Default::default()
     };
-    use OpenocdTclBlock::{AssertRegEq, Echo, PollUntilRegEq, WriteReg};
 
+    let mut stdout = io::stdout();
+    let result = console
+        .interact(&*uart, None, Some(&mut stdout))
+        .context("failed to interact with console")?;
+
+    match result {
+        ExitStatus::ExitSuccess => Ok(()),
+        result => bail!("FAIL: {result:?}"),
+    }
+}
+
+/// Iteration 2 of the `rom_bootstrap_rma` testpoint.
+///
+/// Issues the lifecycle RMA command and ensures the RMA transition is completed
+/// successfully.
+fn test_rma_command(opts: &Opts, transport: &TransportWrapper) -> anyhow::Result<()> {
+    let rma_bootstrap_strapping = transport.pin_strapping("RMA_BOOTSTRAP")?;
+    let tap_strapping = transport.pin_strapping("PINMUX_TAP_LC")?;
+
+    // Need to reset with `RMA_BOOTSTRAP` set.
+    rma_bootstrap_strapping
+        .apply()
+        .context("failed to apply RMA_BOOTSTRAP strapping")?;
+
+    // We keep `PINMUX_TAP_LC` set so that resets due to transition don't
+    // disconnect us from the LC TAP.
+    tap_strapping
+        .apply()
+        .context("failed to apply PINMUX_TAP_LC strapping")?;
+
+    transport
+        .reset_target(opts.init.bootstrap.options.reset_delay, true)
+        .context("failed to reset target")?;
+
+    // Remove the RMA_BOOTSTRAP strapping to bring the device out of the RMA spin loop.
+    rma_bootstrap_strapping
+        .remove()
+        .context("failed to remove strapping RMA_BOOTSTRAP")?;
+
+    // NOTE: without this sleep, the transition starts but isn't successful.
+    // The sleep must happen anywhere before the `START` `TRANSITION_CMD` is given.
+    // Not sure why at the moment, seems like the chip isn't ready even though the
+    // `STATUS.READY` bit is checked.
+    std::thread::sleep(Duration::from_secs(1));
+
+    log::info!("Connecting to JTAG interface");
+
+    let jtag = transport.jtag(&opts.jtag).context("failed to get JTAG")?;
+    jtag.connect(JtagTap::LcTap)
+        .context("failed to connect to JTAG")?;
+
+    // Wait for the lifecycle controller to enter the `READY` state from
+    // which it accepts transition commands.
+    wait_until_ready(&jtag, Duration::from_secs(3))
```

Comment:
you could probably replace a good majority of this code with the `trigger_lc_transition` [testutil](https://cs.opensource.google/opentitan/opentitan/+/master:sw/host/opentitanlib/src/test_utils/lc_transition.rs;drc=6994537bbe708134d423c270c2c4c2b2500ceda1;l=42) now. Note you will have to update it to accept a transition token

Reply:
Will be adding conditional transition support for [manuf_cp_unlock_raw](https://github.com/lowRISC/opentitan/pull/17864) (unless someone beats me to it) at which point I'll update this test, too.